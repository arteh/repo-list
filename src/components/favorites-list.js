import React from 'react';

export const FavoritesList = () => {
    return <div>favorites list</div>
}