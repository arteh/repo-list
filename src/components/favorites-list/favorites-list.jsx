import React, { useState, useEffect } from "react";
import { AiFillHeart, AiOutlineStar } from "react-icons/ai";

import "../../components/list.css";

export const FavoritesList = () => {
  const [faveList, setFaveList] = useState([]);

  useEffect(() => {
    let items = [],
      keys = Object.keys(localStorage).filter((key) =>
        key.includes("favorite")
      ),
      i = keys.length;
    while (i--) {
      items.push(JSON.parse(localStorage.getItem(keys[i])));
    }

    if (items) {
      setFaveList(items);
    }
  }, []);

  return (
    <div>
      <h1>My favorite github repositories</h1>

      {faveList
        ? faveList?.map((item, index) => (
            <div className="item" key={`fave-index-${index}`}>
              <div className="name">{item?.name}</div>
              <div className="description">{item?.description}</div>
              <div className="stars">
                {item?.stargazers_count} <AiOutlineStar />
              </div>

              <div className="favorite-btn">
                <button type="button">
                  <AiFillHeart fill="#EF8354" />
                </button>
              </div>
            </div>
          ))
        : null}
    </div>
  );
};
