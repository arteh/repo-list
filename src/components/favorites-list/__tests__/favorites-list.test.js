import React from "react";
import { screen, render, cleanup } from "@testing-library/react";
import { FavoritesList } from "../favorites-list";

afterEach(cleanup);

describe("Favorites List component", () => {
  it("should render the correct title", () => {
    render(<FavoritesList />);
    expect(screen.getByText("My favorite github repositories")).toBeInTheDocument();
  });

  it("should render the correct page elements", () => {
    const { asFragment } = render(<FavoritesList />);
    expect(asFragment()).toMatchSnapshot();
  });
});