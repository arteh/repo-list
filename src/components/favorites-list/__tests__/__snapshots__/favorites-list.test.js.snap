// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites List component should render the correct page elements 1`] = `
<DocumentFragment>
  <div>
    <h1>
      My favorite github repositories
    </h1>
  </div>
</DocumentFragment>
`;
